AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -174,6 +174,15 @@ @interface ASCollectionView () <ASRangeControllerDataSource, ASRangeControllerDe
    * The collection view never queried your data source before the update to see that it actually had 0 items.
    */
   BOOL _superIsPendingDataLoad;
+
+  /**
+   * It's important that we always check for batch fetching at least once, but also
+   * that we do not check for batch fetching for empty updates (as that may cause an infinite
+   * loop of batch fetching, where the batch completes and performBatchUpdates: is called without
+   * actually making any changes.) So to handle the case where a collection is completely empty
+   * (0 sections) we always check at least once after each update (initial reload is the first update.)
+   */
+  BOOL _hasEverCheckedForBatchFetchingDueToUpdate;
```

Comment:
@Adlai-Holler in a future factoring, this suggests we may want to put Fetching logic in some kind of encapsulation (like an ASBatchFetcher object) to maintain this state and simplify calls to check — so we can use it in table / collection the same way, and maybe offer it as a low-level API for custom scroll view implementers.